pyquil/operator_estimation.py

```diff
@@ -878,31 +894,12 @@ def measure_observables(qc: QuantumComputer, tomo_experiment: TomographyExperime
         for qubit, op_str in max_weight_out_op:
             total_prog += _local_pauli_eig_meas(op_str, qubit)
 
-        # 2. Symmetrization
         qubits = max_weight_out_op.get_qubits()
 
-        if symmetrize_readout == 'exhaustive' and len(qubits) > 0:
-            bitstrings, d_qub_idx = _exhaustive_symmetrization(qc, qubits, n_shots, total_prog)
-
-        elif symmetrize_readout is None and len(qubits) > 0:
-            total_prog_no_symm = total_prog.copy()
-            ro = total_prog_no_symm.declare('ro', 'BIT', len(qubits))
-            d_qub_idx = {}
-            for i, q in enumerate(qubits):
-                total_prog_no_symm += MEASURE(q, ro[i])
-                # Keep track of qubit-classical register mapping via dict
-                d_qub_idx[q] = i
-            total_prog_no_symm.wrap_in_numshots_loop(n_shots)
-            total_prog_no_symm_native = qc.compiler.quil_to_native_quil(total_prog_no_symm)
-            total_prog_no_symm_bin = qc.compiler.native_quil_to_executable(total_prog_no_symm_native)
-            bitstrings = qc.run(total_prog_no_symm_bin)
-
-        elif len(qubits) == 0:
-            # looks like an identity operation
-            pass
-
-        else:
-            raise ValueError("Readout symmetrization method must be either 'exhaustive' or None")
+        # we don't need to do any actual measurement if the combined operator is simply the
+        # identity, i.e. weight=0. We handle this specially below.
+        if len(qubits) > 0:
+            bitstrings = qc.run_symmetrized_readout(total_prog, n_shots, symm_type, qubits)
```

Comment:
Does this mean that `len(qubits) == 0` implies `all(is_identity(setting.out_operator) for setting in settings)` ?

Reply:
Not necessarily. I was just checking my understanding.